src/ILVerify/src/ILImporter.StackValue.cs

```diff
@@ -247,6 +247,232 @@ public static TypeDesc GetIntermediateType(TypeDesc type)
             }
         }
 
+        /// <summary>
+        /// Merges two stack values to a common stack value as defined in the ECMA-335 
+        /// standard III.1.8.1.3 (Merging stack states).
+        /// </summary>
+        /// <param name="valueA">The value to be merged with <paramref name="valueB"/>.</param>
+        /// <param name="valueB">The value to be merged with <paramref name="valueA"/>.</param>
+        /// <param name="merged">The resulting type of merging <paramref name="valueA"/> and <paramref name="valueB"/>.</param>
+        /// <returns>True if merge operation was successful, false if the merge operation failed.</returns>
+        public static bool TryMergeStackValues(StackValue valueA, StackValue valueB, out StackValue merged)
+        {
+            merged = valueA;
+
+            if (valueB.IsReadOnly)
+                merged.SetIsReadOnly();
+
+            // Same type
+            if (valueA.Kind == valueB.Kind && valueA.Type == valueB.Type)
+                return true;
+
+            // One of the two types is a (unboxed) generic parameter (only valid if a == b)
+            if ((valueA.Type != null && valueA.Type.IsGenericParameter && valueA.Kind != StackValueKind.ObjRef) ||
+                (valueB.Type != null && valueB.Type.IsGenericParameter && valueB.Kind != StackValueKind.ObjRef))
```

Comment:
Is this check necessary? These conditions seems to be naturally handled by the code that follows.